File: 백준/Silver/10815. 숫자 카드/숫자 카드.py
N = int(input())
card = list(map(int,input().split()))
card.sort()

M = int(input())
card2 = list(map(int,input().split()))

ans = [0]*M

for i in range(len(card2)):
    start = 0
    end = N-1
    num = (end+start)//2
    while start<=end:
        if card2[i] == card[num]:
            ans[i] = 1
            break
        elif card2[i] < card[num]:
            end = num-1
        else:
            start = num+1
        num = (end+start)//2

print(*ans,sep = ' ')

File: 백준/Silver/10815. 숫자 카드/숫자 카드.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();

        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());

        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(arr);

        int m = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < m; i++) {
            int num = Integer.parseInt(st.nextToken());
            int start = 0;
            int end = n - 1;
            boolean check = false;
            while(start <= end){
                int mid = (start + end) / 2;
                if(num == arr[mid]){
                    sb.append(1 + " ");
                    check = true;
                    break;
                }else if(num < arr[mid]){
                    end = mid - 1;
                }else{
                    start = mid + 1;
                }
            }
            if(!check) sb.append(0 + " ");
        }
        System.out.println(sb);
    }
}